opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -263,7 +263,94 @@ type LogsData = Box<(LogRecord, InstrumentationScope)>;
 /// let provider = LoggerProvider::builder()
 ///     .with_log_processor(processor)
 ///     .build();
+/// ```
+///
+/// **Memory Management in BatchLogProcessor**
+///
+/// The `BatchLogProcessor` manages memory through the following stages of log processing:
+///
+/// 1. **Record Ingestion**:
+///    - Each `LogRecord` is **cloned** upon entering the processor.
+///    - `LogRecordAttributes` utilize a hybrid memory model:
+///      - First 5 attributes are **stack-allocated**.
+///      - Adding additional attributes trigger **heap allocation** in a dynamically growing vector.
+///    - The `LogRecord` and its associated `InstrumentationScope` are **boxed together**
+///      to allocate them on the heap before entering the queue. This means:
+///      - The `LogRecord`'s inline attributes (if any) are moved to the heap as part of the boxed structure.
+///      - Any dynamically allocated data already on the heap (e.g., strings, overflow attributes) remains unaffected.
+///      - Ownership of the boxed data is transferred to the queue, ensuring it can be processed independently of the original objects.
```

Comment:
Not sure I follow this.. The data inside box is already a clone, so it is already independent of originals right?